### <p style="text-align: right;"> &#9989; Team member names </p>

# Project title and description

**Lorem Ipsum** is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.

Contrary to popular belief, **Lorem Ipsum** is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source. Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics, very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in section 1.10.32.

---
#### Importing libraries

In [1]:
import pandas as pd

#### Download and save file to current directory

In [2]:
!curl -o "FoodEnvironmentAtlas.xls" https://www.ers.usda.gov/webdocs/DataFiles/80526/FoodEnvironmentAtlas.xls?v=7810.2

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.7M  100 13.7M    0     0  2592k      0  0:00:05  0:00:05 --:--:-- 3001k


#### Read downloaded file from its path into a variable - data

In [2]:
file = "./FoodEnvironmentAtlas.xls"
data = pd.ExcelFile(file)

#### Read all sheets into a dictionary of dataframes 

In [3]:
sheet_dict = {sheet_name: data.parse(sheet_name) for sheet_name in data.sheet_names}

#### Iterate over all sheets and print the name, number of rows and columns of each sheet

In [11]:
for sheet_name, df in sheet_dict.items():
    print("Sheet Name: {}".format(sheet_name.strip()))
    rows, columns = df.shape
    print("Number of rows and columns are {} and {}".format(rows, columns))
    print()

Sheet Name: Read_Me
Number of rows and columns are 17 and 1

Sheet Name: Variable List
Number of rows and columns are 281 and 7

Sheet Name: Supplemental Data - County
Number of rows and columns are 3142 and 12

Sheet Name: Supplemental Data - State
Number of rows and columns are 51 and 44

Sheet Name: ACCESS
Number of rows and columns are 3143 and 44

Sheet Name: STORES
Number of rows and columns are 3143 and 39

Sheet Name: RESTAURANTS
Number of rows and columns are 3143 and 19

Sheet Name: ASSISTANCE
Number of rows and columns are 3143 and 57

Sheet Name: INSECURITY
Number of rows and columns are 3143 and 9

Sheet Name: TAXES
Number of rows and columns are 3143 and 8

Sheet Name: LOCAL
Number of rows and columns are 3144 and 100

Sheet Name: HEALTH
Number of rows and columns are 3143 and 14

Sheet Name: SOCIOECONOMIC
Number of rows and columns are 3143 and 18



#### Extract the store availability variables 

In [12]:
store_availability = sheet_dict['STORES']

---
## Exploration Data Analysis (EDA)

#### Check the dimension of the data set

In [14]:
row, columns = store_availability.shape
print("Number of rows and columns are {} and {}".format(rows, columns))

Number of rows and columns are 3143 and 39


In [22]:
store_availability.head()

FIPS State   County  GROC11  GROC16  PCH_GROC_11_16  GROCPTH11  GROCPTH16  \
0  1001    AL  Autauga       5       3      -40.000000   0.090581   0.054271   
1  1003    AL  Baldwin      27      29        7.407407   0.144746   0.139753   
2  1005    AL  Barbour       6       4      -33.333333   0.219370   0.155195   
3  1007    AL     Bibb       6       5      -16.666667   0.263794   0.220916   
4  1009    AL   Blount       7       5      -28.571429   0.121608   0.086863   

   PCH_GROCPTH_11_16  SUPERC11  ...  PCH_SNAPS_12_17  SNAPSPTH12  SNAPSPTH17  \
0         -40.085748         1  ...        19.376392    0.674004    0.804747   
1          -3.449328         6  ...        36.927711    0.725055    0.890836   
2         -29.254287         0  ...         3.349282    1.280590    1.424614   
3         -16.254289         1  ...        11.794872    0.719122    0.801423   
4         -28.571429         1  ...         5.701754    0.657144    0.692374   

   PCH_SNAPSPTH_12_17  WICS11  WICS16  PCH_WICS_11_16  WICSPTH11  WICSPTH16  \
0           19.397900     5.0     5.0        0.000000   0.090567   0.090511   
1           22.864524    26.0    28.0        7.692307   0.139380   0.134802   
2           11.246689     7.0     6.0      -14.285714   0.255942   0.232387   
3           11.444711     6.0     5.0      -16.666666   0.263771   0.221474   
4            5.361034     8.0     8.0        0.000000   0.139000   0.139089   

   PCH_WICSPTH_11_16  
0          -0.061543  
1          -3.284727  
2          -9.203081  
3         -16.035471  
4           0.064332  

[5 rows x 39 columns]

*The dimension of the data is **3143** by **39**; the variable names are consistent with the descrption in the variable list sheet. The first few rows indicates I have the right data.*

In [31]:
store_availability.isna().sum()

FIPS                     0
State                    0
County                   0
GROC11                   0
GROC16                   0
PCH_GROC_11_16          21
GROCPTH11                0
GROCPTH16                0
PCH_GROCPTH_11_16       16
SUPERC11                 0
SUPERC16                 0
PCH_SUPERC_11_16       133
SUPERCPTH11              0
SUPERCPTH16              0
PCH_SUPERCPTH_11_16    132
CONVS11                  0
CONVS16                  0
PCH_CONVS_11_16         26
CONVSPTH11               0
CONVSPTH16               0
PCH_CONVSPTH_11_16      21
SPECS11                  0
SPECS16                  0
PCH_SPECS_11_16        250
SPECSPTH11               0
SPECSPTH16               0
PCH_SPECSPTH_11_16     247
SNAPS12                  0
SNAPS17                 26
PCH_SNAPS_12_17         32
SNAPSPTH12               0
SNAPSPTH17              27
PCH_SNAPSPTH_12_17      33
WICS11                 132
WICS16                 158
PCH_WICS_11_16         159
WICSPTH11              135
W